backend_new/src/services/application-flagged-set.service.ts

```diff
@@ -0,0 +1,951 @@
+
+    if (!rawAfs) {
+      throw new NotFoundException(
+        `applicationFlaggedSet ${afsId} was requested but not found`,
+      );
+    }
+
+    return true;
+  }
+
+  /**
+    this goes through listings that have had an application added since the last cronjob run
+    it calls a series of helpers to add to or build a flagged set if duplicates are found
+  */
+  async process(): Promise<SuccessDTO> {
+    this.logger.warn('running the Application flagged sets cron job');
+    await this.markCronJobAsStarted();
+    const outOfDateListings = await this.prisma.listings.findMany({
+      select: {
+        id: true,
+        afsLastRunAt: true,
+      },
+      where: {
+        lastApplicationUpdateAt: {
+          not: null,
+        },
+        AND: [
+          {
+            OR: [
+              {
+                afsLastRunAt: {
+                  equals: null,
+                },
+              },
+              {
+                afsLastRunAt: {
+                  lte: this.prisma.listings.fields.lastApplicationUpdateAt,
+                },
+              },
+            ],
+          },
+        ],
+      },
+    });
+    this.logger.warn(
+      `updating the flagged sets for ${outOfDateListings.length} listings`,
+    );
+
+    for (const listing of outOfDateListings) {
+      const newApplications = await this.prisma.applications.findMany({
+        where: {
+          listingId: listing.id,
+          updatedAt: {
+            gte: listing.afsLastRunAt,
+          },
+        },
+        include: {
+          applicant: true,
+          householdMember: true,
+        },
+      });
+
+      for (const application of newApplications) {
+        await this.testApplication(
+          mapTo(Application, application),
+          application.listingId,
+        );
+      }
+      await this.prisma.listings.update({
+        where: {
+          id: listing.id,
+        },
+        data: {
+          afsLastRunAt: new Date(),
+        },
+      });
+    }
+
+    return {
+      success: true,
+    };
+  }
+
+  /**
+    marks the db record for this cronjob as begun or creates a cronjob that
+    is marked as begun if one does not already exist 
+  */
+  async markCronJobAsStarted(): Promise<void> {
+    const job = await this.prisma.cronJob.findFirst({
+      where: {
+        name: CRON_JOB_NAME,
+      },
+    });
+    if (job) {
+      // if a job exists then we update db entry
+      await this.prisma.cronJob.update({
+        data: {
+          lastRunDate: new Date(),
+        },
+        where: {
+          id: job.id,
+        },
+      });
+    } else {
+      // if no job we create a new entry
+      await this.prisma.cronJob.create({
+        data: {
+          lastRunDate: new Date(),
+          name: CRON_JOB_NAME,
+        },
+      });
+    }
+  }
+
+  /**
+    tests application to see if its a duplicate
+    if it is then its either added to a flagged set or a new flagged set is created
+  */
+  async testApplication(
```

Comment:
question: I think this all checks out, but could you spend some time thinking about how this function could be made more readable?

Reply:
Hey sean, I'm not really sure how to condense or pull this function apart further. I thought about making some more helpers but I feel like it might make this file/function a bit "spaghetti-y" with having to jump around to figure out whats its trying to do 

I made some small tweaks, but lemme know if you have anything in mind to really clean this up

Reply:
Thanks, @YazeedLoonat ,I'm not sure I do without making it harder to follow in other ways like you're saying (at least not without spending more time).